### PR TITLE
BE generate board

### LINKED ISSUE #2: /board/generate
GET - /generateBoard

Query: {
  difficulty: integer
}

Difficulty is like minecraft difficulty. 1 is easy, 2 is normal, 3 is hard.

Returns a 2D-array: Board, representing a minesweeper board.

### PR DESCRIPTION
**Pull request by Suchpuns**
Created a route in the backend:
GET: board/generate

Returns a new board given
{
  difficulty: integer
}
in the query.

### CODE REVIEW

**Comment by floyrin** [comment]
i think this is incredible!

File: server/src/routes/board.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * 2: hard = 20, 99
+ */
+router.get("/generate", (req: Request, res: Response) => {
+  const boardSizes = [
+    { x: 8, y: 10, bombNum: 10 },
+    { x: 14, y: 40, bombNum: 40 },
+    { x: 20, y: 99, bombNum: 99 },
+  ];
+  if (req.query.difficulty === undefined) {
+    return res.status(400).send({ error: "Missing difficulty in query" });
```
Comment (floyrin) [inline server/src/routes/board.ts]: where's your full stop??

File: server/src/routes/board.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+/**
+ * Updates the cell values next to a bomb
+ * @param board The 2D array of the board
+ * @param x     x position of the bomb
+ * @param y     y position of the bomb
+ */
+function updateValues(board: Cell[][], x: number, y: number): void {
+  for (let i = x - 1; i <= x + 1; i++) {
+    for (let j = y - 1; j <= y + 1; j++) {
+      if (
```
Comment (floyrin) [inline server/src/routes/board.ts]: bro?? this could be one line so true

File: server/src/routes/board.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+function printBoard(board: Cell[][]): void {
+  let row: string = "";
+  for (let i = 0; i < board.length; i++) {
+    row = "";
+    for (let j = 0; j < board[0].length; j++) {
+      row += " \t" + board[i][j].value.toString();
+    }
+    console.log(row);
```
Comment (floyrin) [inline server/src/routes/board.ts]: sussy console.log

File: server/tsconfig.json [hunk 4]
```diff
@@ -47,7 +47,7 @@
     // "declaration": true,                              /* Generate .d.ts files from TypeScript and JavaScript files in your project. */
     // "declarationMap": true,                           /* Create sourcemaps for d.ts files. */
     // "emitDeclarationOnly": true,                      /* Only output d.ts files and not JavaScript files. */
-    // "sourceMap": true,                                /* Create source map files for emitted JavaScript files. */
+    "sourceMap": true,                                /* Create source map files for emitted JavaScript files. */
```
Comment (floyrin) [inline server/tsconfig.json:50]: so true

**Review by floyrin — CHANGES_REQUESTED** [review]

File: server/src/routes/board.ts (hunk 3 above)
Comment (Suchpuns) [reply server/src/routes/board.ts]: oops

**Review by floyrin — APPROVED** [review]

**Comment by floyrin** [comment]
image

**Pull request merged**